src/data/fin/tuple/nat_antidiagonal.lean

```diff
@@ -0,0 +1,180 @@
+/-
+Copyright (c) 2022 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+
+import data.fin.vec_notation
+import algebra.big_operators.basic
+import data.list.nat_antidiagonal
+import data.multiset.nat_antidiagonal
+import data.finset.nat_antidiagonal
+import logic.equiv.fin
+
+/-!
+# Collections of tuples of naturals with the same sum
+
+This file generalizes `list.nat.antidiagonal n`, `multiset.nat.antidiagonal n`, and
+`finset.nat.antidiagonal n` from the pair of elements `x : ℕ × ℕ` such that `n = x.1 + x.2`, to
+the sequence of elements `x : fin k → ℕ` such that `n = ∑ i, x i`.
+
+## Main definitions
+
+* `list.nat.antidiagonal_tuple`
+* `multiset.nat.antidiagonal_tuple`
+* `finset.nat.antidiagonal_tuple`
+
+## Main results
+
+* `antidiagonal_tuple 2 n` is analogous to `antidiagonal n`:
+
+  * `list.nat.antidiagonal_tuple_two`
+  * `multiset.nat.antidiagonal_tuple_two`
+  * `finset.nat.antidiagonal_tuple_two`
+
+## Implementation notes
+
+While we could implement this by filtering `(fintype.pi_finset $ λ _, range (n + 1))` or similar,
+this implementation would be much slower.
+-/
+
+open_locale big_operators
+
+/-! ### Lists -/
+
+namespace list.nat
+
+/-- `list.antidiagonal_tuple k n` is a list of `k`-tuples summing to `n`.
```

Comment:
Please add to this docstring that (i) the list contains all such tuples, (ii) it is nodup, and (iii) maybe explain how it's sorted.